### PR TITLE
Add : https_for_ca_consumer property to rhsm_register resource

### PR DESCRIPTION
**Pull request by jasonwbarnett**
## Description

This will enable people to increase their security posture by verifying the remote satellite host SSL certificate.

## Related Issue

N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/resource/rhsm_register.rb [hunk 1]
```diff
@@ -60,6 +60,10 @@ class RhsmRegister < Chef::Resource
         description: "If true, the system will be registered even if it is already registered. Normally, any register operations will fail if the machine has already been registered.",
         default: false, desired_state: false
 
+      property :use_https, [TrueClass, FalseClass],
+        description: "If true, chef will fetch the katello-ca-consumer-latest.noarch.rpm from the satellite_host using HTTPS.",
+        default: false, desired_state: false
```
Comment (lamont-granquist) [inline lib/chef/resource/rhsm_register.rb]: should probably flip the default to true.  this is also a very generic sounding name of a property for a very specific function.  it should probably be `use_katello_https` or something like that.

Comment (lamont-granquist) [reply lib/chef/resource/rhsm_register.rb]: also maybe consider just fixing this to be hardcoded?  would anyone really not want this?

Comment (jasonwbarnett) [reply lib/chef/resource/rhsm_register.rb]: Out of the box internal Satellite servers use a self-signed certificate. Red Hat themselves even reference http (as opposed to https) urls for installation of katello-ca-consumer RPM.

Reference: https://access.redhat.com/documentation/en-us/red_hat_satellite/6.6/html-single/quick_start_guide/index#registering_existing_system

Comment (jasonwbarnett) [reply lib/chef/resource/rhsm_register.rb]: > should probably flip the default to true. this is also a very generic sounding name of a property for a very specific function. it should probably be `use_katello_https` or something like that.

Agreed. I can't think of a short name that gives a clear gist of what it is for that isn't huge. Also, it's not katello, it's the katello-ca-consumer package which are different things.

Comment (lamont-granquist) [reply lib/chef/resource/rhsm_register.rb]: `https_for_ca_consumer true` ?

Comment (jasonwbarnett) [reply lib/chef/resource/rhsm_register.rb]: > `https_for_ca_consumer true` ?

works for me. I'll update and re-push in a bit.

File: lib/chef/resource/rhsm_register.rb [hunk 2]
```diff
@@ -60,6 +60,10 @@ class RhsmRegister < Chef::Resource
         description: "If true, the system will be registered even if it is already registered. Normally, any register operations will fail if the machine has already been registered.",
         default: false, desired_state: false
 
+      property :https_for_ca_consumer, [TrueClass, FalseClass],
+        description: "If true, chef will fetch the katello-ca-consumer-latest.noarch.rpm from the satellite_host using HTTPS.",
```
Comment (tas50) [inline lib/chef/resource/rhsm_register.rb]: The use of Chef needs to be a dist value here for PRODUCT

**Pull request merged**